specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/AzureADAdministrator.json

```diff
@@ -0,0 +1,349 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "MySQLManagementClient",
+    "description": "The Microsoft Azure management API provides create, read, update, and delete functionality for Azure MySQL resources including servers, databases, firewall rules and configurations.",
+    "version": "2021-12-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/flexibleServers/{serverName}/administrators/{administratorName}": {
+      "put": {
+        "tags": [
+          "AzureADAdministrators"
+        ],
+        "operationId": "AzureADAdministrators_CreateOrUpdate",
+        "x-ms-examples": {
+          "Create an azure ad administrator": {
+            "$ref": "./examples/AzureADAdministratorCreate.json"
+          }
+        },
+        "description": "Creates or updates an existing Azure Active Directory administrator.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AdministratorNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AzureADAdministrator"
+            },
+            "description": "The required parameters for creating or updating an aad administrator."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AzureADAdministrator"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/AzureADAdministrator"
+            }
+          },
+          "202": {
+            "description": "Accepted"
+          },
```

Comment:
202 is an old model for async PUT, please see [https://github.com/Azure/azure-resource-manager-rpc/blob/a7dd5c0021d0a84bcad6d4736e3fcc3b319b20af/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/a7dd5c0021d0a84bcad6d4736e3fcc3b319b20af/v1.0/async-api-reference.md#creatingupdating-using-put)

Reply:
Thanks @zizw123  for this information. But to change the old async PUT model to new, we need a very big change to refactor the controller of our Resource Provide, because all our APIs are using the same pattern now. Since this is a part of existing api-version, we have to use the existing model. 
And we will evaluate the effort in the team and put the new model change to our backlog items.